data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/JavaClientAccumulatingUncompressedBulkRequest.java

```diff
@@ -74,6 +76,10 @@ private long estimateBulkOperationSize(BulkOperationWrapper bulkOperation) {
         if (anyDocument == null)
             return OPERATION_OVERHEAD;
 
+        if (anyDocument instanceof JsonNode) {
+            return OPERATION_OVERHEAD + ((JsonNode)anyDocument).toString().length();
```

Comment:
At this point, we are serializing the JsonNode into a string. So is there any real gain in keeping the `JsonNode`?

It seems we can simplify by using the string and then converting to a `JsonNode` when needed for the `update` condition.

Reply:
we need getJsonNode() api to pass to the update operation. I don't think reconstructing it from Json String would be efficient. And we need toJsonString() to get the size. We can do different ways but I think we will need both these